File: doc/src_OpenGL/Furniture.cpp
// Furniture.cpp: implementation of the Furniture class.
//
//////////////////////////////////////////////////////////////////////
#include "Stdafx.h"
#include "OpenGL.h"

#include "Inkludy.h"

#ifdef _DEBUG
#undef THIS_FILE
static char THIS_FILE[]=__FILE__;
#define new DEBUG_NEW
#endif

//////////////////////////////////////////////////////////////////////
// Construction/Destruction
//////////////////////////////////////////////////////////////////////

Furniture::Furniture(Position* p, int weight)
: WorldObject(WorldObjectVerifier::FURNITURE,p,false,false,true,true)
{	
	if( weight < 0 )
		this->weight = 1;
	else 
		this->weight = weight;
	direction = Direction::randDirection();
	type = rand() % 4;
	
}

Furniture::~Furniture()
{

}
void Furniture::evolve()
{

}

bool Furniture::conditionalMovement(WorldObject* const worldObject, const int direction,
		const int MaxPower,	int& usedPower)
{
	usedPower += weight;
	return getWorld()->move(this,direction,MaxPower,usedPower);
}

int Furniture::getDirection()
{
	return direction;
}

int Furniture::getType()
{
	return type;
}

File: app/src/main/java/robot/object/Furniture.java
package robot.object;

import org.apache.commons.lang.mutable.MutableInt;
import robot.Direction;
import robot.Position;
import robot.WorldObjectVerifier;

import java.util.Random;

public class Furniture extends WorldObject {

    private int weight;
    private int direction;
    private int type;

    public Furniture(Position p, int weight) {
        super(WorldObjectVerifier.FURNITURE.getIntValue(),p,false,false,true,true);
        if( weight < 0 ) {
            this.weight = 1;
        }
        else {
            this.weight = weight;
        }
        direction = Direction.randDirection();
        type = new Random().nextInt(4);

    }

    public void evolve() {

    }

    public boolean conditionalMovement(final WorldObject worldObject, final int direction,final int MaxPower, MutableInt usedPower) {
        usedPower.add(weight);
        return getWorld().move(this,direction,MaxPower,usedPower);
    }

    public int getDirection()
    {
        return direction;
    }

    public int getType()
    {
        return type;
    }

}